contrib/mercurial/template.py

```diff
@@ -0,0 +1,26 @@
+pkgname = "mercurial"
+pkgver = "6.7.3"
+pkgrel = 0
+build_style = "python_pep517"
+hostmakedepends = [
+    "python-build",
+    "python-installer",
+    "python-setuptools",
+    "python-wheel",
+]
+makedepends = [
+    "gettext-devel",
+    "musl-devel",
+    "python-devel",
+    "zlib-devel",
+    "zstd-devel",
+]
+checkdepends = ["python-pytest"]
+pkgdesc = "Distributed source control management"
+maintainer = "ttyyls <contact@behri.org>"
+license = "GPL-2.0-or-later"
+url = "https://www.mercurial-scm.org"
+source = f"https://www.mercurial-scm.org/release/mercurial-{pkgver}.tar.gz"
+sha256 = "00196944ea92738809317dc7a8ed7cb21287ca0a00a85246e66170955dcd9031"
+# vendored dependencies tests are broken on python 3.11 and up
```

Comment:
could probably still be made to work with enough checkdepends

Reply:
no they use python's imp which was part of the stdlib until 3.11 where it was deprecated. the vendored version of zstd here and upstream both rely on it for all their tests. skipping zstd tests from mercurial causes everything else to be skipped so i thought to just !check it

Reply:
that means you didn't run them correctly because that 'imp' error is inside the vendored zstd tests.. which are not related to the actual mercurial tests which can only be ran with ./tests/run-tests.py and fail for completely unrelated reasons